cmd/nebraska/metrics.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	defaultMetricsUpdateInterval = 5 * time.Second
+)
+
+var (
+	appInstancePerChannelGaugeMetric = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "nebraska",
+			Name:      "application_instances_per_channel",
+			Help:      "Number of applications from specific channel running on instances",
+		},
+		[]string{
+			"application",
+			"version",
+			"channel",
+		},
+	)
+
+	failedUpdatesGaugeMetric = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "nebraska",
+			Name:      "failed_updates",
+			Help:      "Number of failed updates of an application",
+		},
+		[]string{
+			"application",
+		},
+	)
+)
+
+// registerNebraskaMetrics registers the application metrics collector with the DefaultRegistrer.
+func registerNebraskaMetrics() error {
+	err := prometheus.Register(appInstancePerChannelGaugeMetric)
+	if err != nil {
+		return err
+	}
+	err = prometheus.Register(failedUpdatesGaugeMetric)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// getMetricsRefreshInterval returns the metrics update Interval key is set in the environment as time.Duration,
+// NEBRASKA_METRICS_UPDATE_INTERVAL. The variable must be a string acceptable by time.ParseDuration
+// If not returns the default update interval.
+func getMetricsRefreshInterval() time.Duration {
+	refreshIntervalEnvValue := os.Getenv("NEBRASKA_METRICS_UPDATE_INTERVAL")
+	if refreshIntervalEnvValue == "" {
+		return defaultMetricsUpdateInterval
+	}
+
+	refreshInterval, err := time.ParseDuration(refreshIntervalEnvValue)
+	if err != nil {
```

Comment:
Checking for negative duration is missing.
```suggestion
	if err != nil || refreshInterval <= 0 {
```

Reply:
Or consider having a separate `if` for checking for negative value with an appriopriate error message.